pkg/service/signal.go

```diff
@@ -139,26 +139,34 @@ func (r *signalService) RelaySignal(stream psrpc.ServerStream[*rpc.RelaySignalRe
 		"connID", ss.ConnectionId,
 	)
 
-	reqChan := routing.NewDefaultMessageChannel(livekit.ConnectionID(ss.ConnectionId))
-	defer reqChan.Close()
-
 	sink := routing.NewSignalMessageSink(routing.SignalSinkParams[*rpc.RelaySignalResponse, *rpc.RelaySignalRequest]{
 		Logger:       l,
 		Stream:       stream,
 		Config:       r.config,
 		Writer:       signalResponseMessageWriter{},
 		ConnectionID: livekit.ConnectionID(ss.ConnectionId),
 	})
+	reqChan := routing.NewDefaultMessageChannel(livekit.ConnectionID(ss.ConnectionId))
+
+	go func() {
+		err = routing.CopySignalStreamToMessageChannel[*rpc.RelaySignalResponse, *rpc.RelaySignalRequest](
+			stream,
+			reqChan,
+			signalRequestMessageReader{},
+			r.config,
+		)
+		l.Infow("signal stream closed", "error", err)
+
+		reqChan.Close()
+	}()
 
 	err = r.sessionHandler(ctx, livekit.RoomName(ss.RoomName), *pi, livekit.ConnectionID(ss.ConnectionId), reqChan, sink)
 	if err != nil {
 		l.Errorw("could not handle new participant", err)
 		return
 	}
 
-	err = routing.CopySignalStreamToMessageChannel[*rpc.RelaySignalResponse, *rpc.RelaySignalRequest](stream, reqChan, signalRequestMessageReader{}, r.config)
-	l.Infow("signal stream closed", "error", err)
-
+	stream.Hijack()
```

Comment:
what does Hijack do?

Reply:
normally when the stream handler function returns the stream is closed by psrpc - when a stream is hijacked it's up to the handler to close it manually